internal/kibana/reconciler_test.go

```diff
@@ -418,6 +420,40 @@ func TestDeploymentDifferentWithProxyEnvVar(t *testing.T) {
 	}
 }
 
+func TestKibanaSecurityContext(t *testing.T) {
+	clusterRequest := &KibanaRequest{
+		client: nil,
+		cluster: &kibana.Kibana{
+			ObjectMeta: metav1.ObjectMeta{
+				Namespace: "test-namespace",
+			},
+			Spec: kibana.KibanaSpec{},
+		},
+	}
+
+	podSpec := newKibanaPodSpec(clusterRequest, "test-app-name", nil, nil, "")
+	expectedPod := v1.PodSecurityContext{
+		RunAsNonRoot: pointer.Bool(true),
+	}
+
+	if diff := cmp.Diff(*podSpec.SecurityContext, expectedPod); diff != "" {
```

Comment:
I'd probably define the `expectedPod` variable as a pointer as well, so that I could pass them both as pointers to the Diff. A `nil` SecurityContext would cause a panic now.